README.md

```diff
@@ -0,0 +1,130 @@
+
+```Javascript
+<style>
+  ::highlight(foo){
+      color: blue;
+  }
+</style>
+<script>
+  let customHighlight = new Highlight();
+  CSS.highlights.set('foo', customHighlight);
+</script>
+```
+
+In the case above, a custom highlight is registered under the name 'foo', so all ranges contained in it will be styled with text color blue, as specified in the style section.
+
+Finally, the next example shows a simple way of highlighting selected text with the Highlight API:
+
+```Javascript
+<style>
+  ::highlight(example-highlight) {
+    color: blue;
+  }
+</style>
+
+<script>
+  const highlightName = "example-highlight";
+
+  function highlightSelectionWithAPI() {
+    let range = window.getSelection().getRangeAt(0);
+    if(CSS.highlights.has("example-highlight")) 
+      CSS.highlights.get(highlightName).add(range);
+    else
+      CSS.highlights.set(highlightName, new Highlight(range));
+  }
+  
+  function clearAPIHighlight() {
+    CSS.highlights.delete(highlightName);
+  }
+</script>
+```
+
+## How to highlight text *without* the Highlight API
+
+Traditionally, a common way to highlight different pieces of text dynamically in a webpage is to just wrap the content with a new `<span>` with a particular class or id that is given a specific style. An example of this technique can be found in [chromium code search](https://source.chromium.org)'s custom find on page, where if you look for text inside a source code file, the site highlights all matching strings with new spans. An actual screenshot is show next:
+
+![Screenshot of real-life example of wrapping text with spans to highlight find-on-page results in Chromium Code Search](resources/custom-find-on-page-spans.png)
+
+However, implementing this can be somewhat cumbersome because we need to manipulate the DOM tree by inserting new objects into it. This can lead to many issues:
+
+- We should be careful not to break other processing of the DOM or cached elements, as the number of children/siblings of a node could change. This means the rest of the webpage should support dynamic changes in the DOM and act accordingly.
+- Some cases are not easy to handle. For example, highlighting a chunk of text that crosses a node boundary.
+- Performance could be impacted due to the DOM growing too much. This could affect operations such as rendering frames or performing other actions needed by the site itself. See [Demo 2](https://github.com/ffiori/highlight-api-demos/tree/add-readme#demo-2-performance-of-both-approaches) for a performance comparison with the Highlight API.
+- Reverting the operation is not easy in non-trivial sites. We should traverse the DOM looking for spans with this specific id or class and delete them while reinserting their content into the same place it was before. Furthermore, this operation could be computationally expensive if the DOM's got too big and there are many highlighted spans to remove.
+

```

Comment:
Another thing we could mention is that the `<span>` method gets very complicated if we have multiple types of highlights we want to apply, especially if the different kinds can overlap. This is related to the point about highlighting chunks of text that cross node boundaries, since in this case we are *creating* node boundaries when we add spans.

Reply:
Thanks for mentioning this case, definitely useful 👍